lib/sr.py

```diff
@@ -40,8 +40,9 @@ def plug_pbds(self, verify=True):
         if verify:
             wait_for(self.all_pbds_attached, "Wait for PDBs attached")
 
-    def vdi_uuids(self, managed=False):
-        return safe_split(self.pool.master.xe('vdi-list', {'sr-uuid': self.uuid, 'managed': managed}, minimal=True))
+    def vdi_uuids(self, managed=False, label=''):
+        return safe_split(self.pool.master.xe('vdi-list', {'sr-uuid': self.uuid, 'managed': managed,
+                                                           'name-label': label}, minimal=True))
```

Comment:
Bug: if label is null, the function returns an empty list.

```python
def vdi_uuids(self, managed=False, name=None):
    args = {
        'sr-uuid': self.uuid, 
        'managed': managed
    }
    if name is not None:
        args['name-label'] = name
    return safe_split(self.pool.master.xe('vdi-list', args, minimal=True))
```

Reply:
I would also name the parameter `name_label` to stay close to `xe`'s parameter names.